odoo/tools/misc.py

```diff
@@ -1011,19 +1011,31 @@ class Collector(Mapping):
         for ``defaultdict(list)``.
     """
     __slots__ = ['_map']
+
     def __init__(self):
         self._map = {}
+
     def add(self, key, val):
         vals = self._map.setdefault(key, [])
         if val not in vals:
             vals.append(val)
+
     def __getitem__(self, key):
         return self._map.get(key, ())
+
     def __iter__(self):
         return iter(self._map)
+
     def __len__(self):
         return len(self._map)
 
+    def discard_any(self, keys):
```

Comment:
Maybe a somewhat clearer naming e.g. `discard_keys_and_values` or somesuch, and name the parameter something else since it can match and discard either? (e.g. "needles" or "excludes")